### PR TITLE
docs: adding a threat model diagram for Ubuntu Frame

### PR DESCRIPTION
**Pull request by mattkae**
To view: https://canonical-mir--3484.com.readthedocs.build/3484/explanation/threat-model/

### CODE REVIEW

**Comment by mattkae** [comment]
@Saviq Like this? 
image

**Comment by Saviq** [comment]
> @Saviq Like this?

I think two were better for OSK, unidirectional content and bidirectional input.

Also app needs input / window management one way, and content the other.

**Comment by mattkae** [comment]
image

**Comment by mattkae** [comment]
image

**Comment by Saviq** [comment]
OK, so bidirectional doesn't look too good. Let's just mark the unidirectional edges with `input` to OSK, `input/contents` back to Mir.

We should jump on a call to whack those out, this is too much overhead :)

File: doc/sphinx/explanation/ubuntu_frame_threat_model.mmd [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    user(User)
+    display(Display)
+    app(App)
+
+    subgraph snap [Snap]
+        vnc_server--input-->shell
+        shell--screen content-->vnc_server
+
+        osk--content-->shell
+        shell<--input-->osk
```
Comment (Saviq) [inline doc/sphinx/explanation/ubuntu_frame_threat_model.mmd:18]:
```suggestion
        osk--content/input-->shell
        shell--input-->osk
```

File: doc/sphinx/explanation/ubuntu_frame_threat_model.mmd [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    end
+    
+    shell--contents/configuration--->display
+    snapd--configuration-->shell
+    display--visuals-->user
+    operator_app--help message-->shell
+    user--io events-->shell
```
Comment (Saviq) [inline doc/sphinx/explanation/ubuntu_frame_threat_model.mmd:25]:
```suggestion
    user--input events-->shell
```

**Review by Saviq — APPROVED** [review]
Couple nits!

**Pull request merged**